### ISSUE TITLE
[BUG] running npm install again on workspace deletes all dependencies

### ISSUE THREAD

**Issue by Coridyn** [npm/cli#1784]
### Current Behavior:

Running `npm install` on a workspace project works correctly the first time.

Running `npm install` on the workspace a second time deletes everything in the workspace's `node_modules` directory.

Running `npm install` a third time says the install is "up to date" even though `node_modules` is empty and nothing is installed.

### Expected Behavior:

It is expected that running `npm install` again on a workspace project should install new dependencies that have been added to package.json files or leave the installed dependencies as-is if there are no changes.

It is unexpected that `npm install` empties out `/node_modules` when running the second time.

It is expected that `npm install` should work similarly for a workspace project as for a regular, non-workspace project.

NOTE: It is possible to run `npm update` to re-install workspace dependencies but it is unexpected that it is ok to run `npm install; npm install` on a regular project but have to run `npm install; npm update` on workspace projects.

### Steps To Reproduce:

Reproduction example available here: https://github.com/Coridyn/npm-workspace-test

* Create workspace project
* Run `npm install` on workspace
  * Note that everything installs correctly
* Run `npm install` again on workspace
  * npm removes all packages
  * Workspace's `node_modules` directory is empty except for `.package-lock.json`
* Run `npm install` again on workspace
   * npm does not install dependencies and says project is up to date
  * Workspace's `node_modules` directory is empty except for `.package-lock.json`

### Environment:

- OS: Windows 10 (19041.450)
- Node: 12.16.1
- npm: 7.0.0-beta.10

---------------------

- OS: Ubuntu 18.04.4 (WSL2)
- Node: 12.16.2
- npm: 7.0.0-beta.10

**Comment by lazharichir** [comment]
Similar thing happening for me. I use the workspaces with `npm 7.0.0-beta.11`.

On the first `npm install` everything works well (symlinks and all).

If I then install a package with `npm install lodash` (for example, the same happens with any package), then the workspace's packages are no longer found at runtime (other modules are, such as _lodash_).

I need to delete `package-lock.json` and `node_modules/` and run again `npm install` to have everything working again.

OS: macOS 10.15.6
Node: v14.10.1
npm: 7.0.0-beta.11

**Comment by gcangussu** [comment]
It also happens when using `npx` here, in addition to the scenarios described above. Example:

1. Run first `npm install` -- everything is installed
2. Run `npx eslint .` -- all workspaces exclusive dependencies are deleted (dependencies of the root package.json are kept)

OS: Linux 5.8.9
Node: 12.18.4
npm: 7.0.0-beta.11
npx: 7.0.0-beta.11

**Comment by ruyadorno** [comment]
Thanks for reporting this @Coridyn @lazharichir @gcangussu 👍 it should be fixed now in the latest RC: `v7.0.0-rc.3`

As usual you can get it with `npm i -g npm@next-7`

Please let us know if you run into any more problems and thanks again for the help 😊

**Comment by neodon** [comment]
This appears to be a problem again.

```
my-package $ node -v
v16.9.1
my-package $ npm -v
7.24.0
my-package $ rm -rf node_modules/ package-lock.json pkg/my-workspace/{package-lock.json,node_modules}
my-package $ npm install

added 679 packages, and audited 682 packages in 26s

96 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
my-package $ cd pkg/my-workspace
my-package/pkg/my-workspace $ npm install
npm WARN old lockfile 
npm WARN old lockfile The package-lock.json file was created with an old version of npm,
npm WARN old lockfile so supplemental metadata must be fetched from the registry.
npm WARN old lockfile 
npm WARN old lockfile This is a one-time fix-up, please be patient...
npm WARN old lockfile 

added 632 packages, and audited 634 packages in 6s

91 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
my-package/pkg/my-workspace $ cd ../..
my-package $ npm install

removed 632 packages, and audited 682 packages in 2s

96 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
```

Running `npm install` in the root of a workspace removes all the dependencies for that workspace.
